autobot-backend/api/playwright_session_scope.py

```
@@ -0,0 +1,150 @@
+    if _declared_session_id(body, request.query_params.get("session_id")):
+        return
+
+    client = getattr(request.client, "host", None) if request.client else None
+    # Caller and path are embedded in the message, not passed as %s arguments:
+    # the flood filter keys on the UNINTERPOLATED template plus call site
+    # (#15774), so arguments would hand every caller one shared 5-per-minute
+    # budget and let a chatty client silence a second unscoped one — the log
+    # could no longer answer "who". user_agent stays an argument: it is
+    # attacker-controlled and unbounded, so it must not enter the key space.
+    logger.warning(
+        _message(request.url.path, client or "unknown", refusal) + " user_agent=%s",
```

Ruff (0.16.3)

[warning] Logging statement uses `+`

(G003)